File: backend/src/main/java/com/penocean/ehs/model/Notice.java
package com.penocean.ehs.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notice {

    private Long id;
    private String category;            // NOTICE / ANNOUNCEMENT / URGENT
    private String title;
    private String content;
    private Long authorUserId;
    private LocalDateTime publishedAt;
    private Boolean pinned;
    private Long viewCount;
    private LocalDateTime expiresAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean deleted;
}

File: frontend/src/types/notice.ts
// Backend: /api/notices
import type { PageResponse } from './common'

export type NoticeCategory = 'NOTICE' | 'ANNOUNCEMENT' | 'URGENT'

export interface NoticeListItem {
  id: number
  category: NoticeCategory
  title: string
  authorUserName: string
  publishedAt: string
  pinned: boolean
  viewCount: number
  expiresAt?: string | null
}

export interface NoticeDetail extends NoticeListItem {
  content: string
  attachmentUrl?: string | null
  attachmentName?: string | null
}

export interface NoticePayload {
  category: NoticeCategory
  title: string
  content: string
  pinned?: boolean
  expiresAt?: string | null
}

export interface NoticeListParams {
  category?: NoticeCategory | ''
  keyword?: string
  page?: number
  size?: number
}

export type NoticePage = PageResponse<NoticeListItem>
